assets/js/translator.js

```diff
@@ -447,12 +447,33 @@ function getPairs() /*: JQueryPromise<any> */ {
             }
         });
 
-        for(var srcLang in pairs) {
-            // Default for new users is first available pair
-            curSrcLang = srcLang;
-            curDstLang = pairs[srcLang][0];
-            break;
+        // Default for new users is first available Browser Preference Language pair
+        var brwsLangs = navigator.languages; // Chrome, Mozilla and Safari
+        var prefLang;
+        $.each(brwsLangs, function(index) {
+            if(languages[brwsLangs[index].substr(0, 2)] !== undefined) {
+                curSrcLang = iso639CodesInverse[brwsLangs[index].substr(0, 2)];
```

Comment:
What if none of the languages are valid source languages? This approach seems like it would leave the application in some crazy state.

Reply:
Oops I for got it here. :)

Reply:
it will move to next step below, set to Locale. :).

Reply:
Nope, See below it will set to locale.

Reply:
Done. :)